api/v1_claim_rewards.go

```diff
@@ -225,65 +206,169 @@ func getValidatorAttestation(args GetValidatorAttestationParams) (*SenderAttesta
-				Signature:      signature,
+	// Get validator attestations in parallel rounds
+	successfulValidators := 0
+	currentIndex := 0
+
+	for successfulValidators < validatorsNeeded && currentIndex < len(shuffled) {
+		// Select up to validatorsNeeded validators for this round
+		var candidateNodes []config.Node
+		var candidateIndices []int
+		candidateOwners := make(map[string]bool) // Track owners in this round
+
+		for i := currentIndex; i < len(shuffled) && len(candidateNodes) < validatorsNeeded; i++ {
+			node := shuffled[i]
+			// Skip if we've already used this owner globally or this specific validator is marked bad
+			if usedOperators[node.OwnerWallet] || badValidators[node.Endpoint] {
+				continue
+			}
+			// Skip if we've already picked this owner in this round
+			if candidateOwners[node.OwnerWallet] {
+				continue
 			}
-			validatorAttestation, err := getValidatorAttestation(getValidatorAttestationParams)
+			candidateNodes = append(candidateNodes, node)
+			candidateIndices = append(candidateIndices, i)
+			candidateOwners[node.OwnerWallet] = true
+		}
 
-			if err != nil {
-				return err
+		if len(candidateNodes) == 0 {
+			break // No more valid candidates
+		}
+
+		// Try to get attestations from all candidates in parallel
+		type validatorResult struct {
+			index       int
+			attestation *SenderAttestation
+			err         error
+		}
+
+		results := make([]validatorResult, len(candidateNodes))
+		validatorGroup := &errgroup.Group{}
+
+		for i, node := range candidateNodes {
+			i, node := i, node // capture loop variables
+			validatorGroup.Go(func() error {
+				getValidatorAttestationParams := GetValidatorAttestationParams{
+					Validator:      node.Endpoint,
+					Claim:          rewardClaim.RewardClaim,
+					UserEthAddress: rewardClaim.RecipientEthAddress,
+					Signature:      signature,
+				}
+
+				attestation, err := getValidatorAttestation(getValidatorAttestationParams)
+				results[i] = validatorResult{
+					index:       candidateIndices[i],
+					attestation: attestation,
+					err:         err,
+				}
+				return nil // Don't fail the group on individual validator errors
+			})
+		}
+
+		// Wait for all validators in this round to complete
+		validatorGroup.Wait()
+
+		// Process results
+		for _, result := range results {
+			node := shuffled[result.index]
+			if result.err != nil {
+				// Mark this validator as bad
+				badValidators[node.Endpoint] = true
+				continue
 			}
-			attestations[i+offset] = *validatorAttestation
-			return nil
-		})
+
+			// Success - add attestation and mark owner as used
+			attestations = append(attestations, *result.attestation)
+			usedOperators[node.OwnerWallet] = true
+			successfulValidators++
+
+			// Stop if we have enough
+			if successfulValidators >= validatorsNeeded {
+				break
+			}
+		}
+
+		// Move to next batch of validators
+		if len(candidateIndices) > 0 {
+			currentIndex = candidateIndices[len(candidateIndices)-1] + 1
```

Comment:
this feels weird to me - why keep the candidate indices around and have this logic at the end? can't we update the current index at the top instead? validators will either succeed or fail... either way, we won't retry them right?

like:
```go
for i := currentIndex; i < len(shuffled) && len(candidateNodes) < validatorsNeeded; i++
```

could be
```go
for currentIndex < len(shuffled) && len(candidateNodes) < validatorsNeeded; currentIndex++
```
right?

and then we remove all this end of loop stuff?